notebooks/Adapter_Whisper_Audio_FineTuning.ipynb

```diff
@@ -0,0 +1,555 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import evaluate\n",
+    "\n",
+    "metric = evaluate.load(\"wer\")\n",
+    "\n",
+    "def compute_metrics(pred):\n",
+    "    pred_ids = pred.predictions\n",
+    "    label_ids = pred.label_ids\n",
+    "\n",
+    "    # replace -100 with the pad_token_id\n",
+    "    label_ids[label_ids == -100] = tokenizer.pad_token_id\n",
+    "\n",
+    "    # we do not want to group tokens when computing the metrics\n",
+    "    pred_str = tokenizer.batch_decode(pred_ids, skip_special_tokens=True)\n",
+    "    label_str = tokenizer.batch_decode(label_ids, skip_special_tokens=True)\n",
+    "\n",
+    "    wer = 100 * metric.compute(predictions=pred_str, references=label_str)\n",
+    "\n",
+    "    return {\"wer\": wer}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Initializing the `Whisper` adapters model\n",
+    "\n",
+    "Here we will setup the adapters `Whisper` model. You can see that the code used to initialize the model looks somewhat identical to code used inside `HuggingFace`. This is because `adapters` acts like a wrapper to the `HuggingFace` library, so users familiar with `HuggingFace` can easily integrate their own code while combining it with `adapters` code. The cool thing here is that we can load the `WhisperConfig` directly from `transformers` and use it for our `Whisper` model specifications. \n",
+    "\n",
+    "From the `adapters` module we then import the `WhisperAdapterModel` and intialize it using the config we previously imported."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from transformers import WhisperConfig\n",
+    "from adapters import WhisperAdapterModel\n",
+    "\n",
+    "config = WhisperConfig.from_pretrained(\n",
+    "    model_name_or_path,\n",
+    ")\n",
+    "model = WhisperAdapterModel.from_pretrained(\n",
+    "    model_name_or_path,\n",
+    "    config=config,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now add an untrained adapter named `whisper_adapter` that we will fine-tune instead of the ``Whisper`` model parameters. Afterwards we include a sequence to sequence language modeling head so we can generate tokens from the ``Whisper`` model.\n",
+    "\n",
+    "Once these two components are added, we leverage the `train_adapter` function to make sure `adapters` knows what adapter weights that need to be updated during training."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import adapters\n",
+    "#initialize the Lora Config to use as an adapter\n",
+    "from adapters import LoRAConfig\n",
+    "\n",
+    "adapters.init(model)\n",
```

Comment:
```suggestion
```

Reply:
This method has no effect on models from the class `AdapterModel`, the `init()` method is only needed if we want to enable adapter support on Hugging Face models that were not created using an `AdapterModel` class